packages/cli/src/commands/seed.ts

```diff
@@ -0,0 +1,498 @@
+  validator: DeployablePayloadOrAddress<Identifiable<SignerValidatorPayload>>;
+  allowList: DeployablePayloadOrAddress<
+    Identifiable<SimpleDenyListPayload> | Identifiable<SimpleAllowListPayload>
+  >;
+  incentives: (
+    | Identifiable<AllowListIncentivePayload>
+    | Identifiable<ERC20IncentivePayload>
+    | Identifiable<ERC20VariableCriteriaIncentivePayload>
+    | Identifiable<ERC20VariableIncentivePayload>
+  )[];
+};
+
+export const ActionClaimantSchema = z.object({
+  signatureType: z.nativeEnum(SignatureType),
+  signature: zAbiItemSchema,
+  fieldIndex: z.number().nonnegative(),
+  targetContract: AddressSchema,
+  chainid: z.number().nonnegative(),
+});
+
+export const ActionStepCriteriaSchema = z.object({
+  filterType: z.nativeEnum(FilterType),
+  fieldType: z.nativeEnum(PrimitiveType),
+  fieldIndex: z.number().nonnegative(),
+  filterData: zHexSchema,
+});
+
+export const ActionStepSchema = z.object({
+  signature: zAbiItemSchema,
+  signatureType: z.nativeEnum(SignatureType),
+  actionType: z.number().optional(),
+  targetContract: AddressSchema,
+  chainid: z.number().nonnegative(),
+  actionParameter: ActionStepCriteriaSchema,
+});
+
+export const EventActionSchema = z.object({
+  type: z.literal('EventAction'),
+  actionClaimant: ActionClaimantSchema,
+  actionSteps: z.array(ActionStepSchema).max(4),
+});
+
+export const SignerValidatorSchema = z.object({
+  type: z.literal('SignerValidator'),
+  signers: z.array(AddressSchema),
+  validatorCaller: AddressSchema,
+});
+
+export const SimpleDenyListSchema = z.object({
+  type: z.literal('SimpleDenyList'),
+  owner: AddressSchema,
+  denied: z.array(AddressSchema),
+});
+
+export const SimpleAllowListSchema = z.object({
+  type: z.literal('SimpleAllowList'),
+  owner: AddressSchema,
+  allowed: z.array(AddressSchema),
+});
+
+export const AllowListIncentiveSchema = z.object({
+  type: z.literal('AllowListIncentive'),
+  allowList: AddressSchema,
+  limit: z.coerce.bigint(),
+});
+
+export const ERC20IncentiveSchema = z.object({
+  type: z.literal('ERC20Incentive'),
+  asset: AddressSchema,
+  mintableMockAsset: z.boolean().default(true),
+  strategy: z.nativeEnum(StrategyType),
+  reward: z.coerce.bigint(),
+  limit: z.coerce.bigint(),
+  manager: AddressSchema.optional(),
+});
+
+export const ERC20VariableIncentiveSchema = z.object({
+  type: z.literal('ERC20VariableIncentive'),
+  asset: AddressSchema,
+  mintableMockAsset: z.boolean().default(true),
+  reward: z.coerce.bigint(),
+  limit: z.coerce.bigint(),
+  manager: AddressSchema,
+});
+
+export const IncentiveCriteriaSchema = z.object({
+  criteriaType: z.nativeEnum(SignatureType),
+  signature: zAbiItemSchema,
+  fieldIndex: z.number().nonnegative(),
+  targetContract: AddressSchema,
+});
+
+export const ERC20VariableCriteriaIncentiveSchema = z.object({
+  type: z.literal('ERC20VariableCriteriaIncentive'),
+  asset: AddressSchema,
+  mintableMockAsset: z.boolean().default(true),
```

Comment:
i think this name would be clearer if it was something like `shouldMintAndAllocate` or `autoMintAndFund`